.github/workflows/release.yml

```diff
@@ -1,20 +1,17 @@
-name: Tests and release
+name: Build and Release
 
 on:
-  pull_request:
-  workflow_dispatch:
-  push:
+  workflow_run:
+    workflows: ["Test"]
     branches:
-    - main
-    tags:
-    - "v*"
-  release:
-    types: [published]
+      - main
+    types:
+      - completed
 
 jobs:
   version:
     runs-on: ubuntu-latest
-    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+    if: github.event.workflow_run.conclusion == 'success' && github.event_name == 'push'
```

Comment:
correct me if i'm wrong, but wouldn't this deploy every time the tests pass?

Reply:
Runs every time tests pass in the main branch, and tests run only after pushing a tag starting with "v*" to main (the rest of the time coverage runs).